### DISCUSSION TITLE
Does it support Pytorch version?

### DISCUSSION THREAD

**Discussion by ChenControl** [python-control/python-control#846] [Q&A]
Hi, I use pytorch to make a simulation of robot control. It uses sensor signal as input to train an optimal driving signal.

I know it is wise to use RL to achieve the target. However, I would like to have a try to use LSTM/GRU to make a NN controller.
The problem is when I use python-control to simulate the next step response, it has to been changed from pytorch tensor to numpy enviroment, and back to pytorch tensor again in the forward function.
This operation called in-place operation could fail the loss.backward, leading to 
"RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation.", which means the pytorch could not trace back to calculate the correct gradient.

I am wondering if the pytorch version of control could fix this problem, or I did sth wrong in other places.
Any suggestion?

Thank you very much and best regards.

Chen

**Comment by henklaak** [comment]
Hi, I am not sure that python-control is the right tool to just run simulations.

This library is more of a collection of mathematical tools to analyze/design control systems in different domains.

Maybe SciPy can be of more help? 
E.g. https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.step.html

Nice info though. AI is our 'competitor' in creating control systems. I am following the developments closely ;-)

KR, Henk van der Laak
